driver/network_driver.go

```diff
@@ -252,8 +290,12 @@ func (d NetworkDriver) Join(request *network.JoinRequest) (*network.JoinResponse
 		return nil, err
 	}
 
+	mac, err := d.GetEndpointMAC(request.EndpointID)
+	if err != nil {
+		return nil, err
+	}
 	// libnetwork doesn't set the MAC address properly, so set it here.
```

Comment:
I just tried it without netns.SetVethMac and docker inspect correctly showed the MAC address. 
Docker [PR#755](https://github.com/docker/libnetwork/pull/755) might be what fixed the libnetwork issue.

A separate PR for removing netns.SetVethMac might be good, since it's possibly a functional change. This PR can be applied after.